index.js

```diff
@@ -53,7 +54,12 @@ function NYC (opts) {
   // require extensions can be provided as config in package.json.
   this.require = arrify(config.require || opts.require)
 
-  this.transform = this._createTransform()
+  this.extensions = arrify(config.extensions || opts.extension)
```

Comment:
Perhaps add `.js` onto `this.extensions`. That way you can use it in `_maybeInstrumentSource()`. Could also map it to ensure all extensions are lowercased.